src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,447 @@
+
+
+    def __str__(self) -> str:
+        """String representation of the proto_obj."""
+        self._check_intent_exist()
+
+        return (f"{self._show_basic_info()}"
+            f"\n\n{self._show_parameters()}"
+            f"\n\n{self._show_training_phrases()}")
+
+
+    def _check_intent_exist(self):
+        """Check if the proto_obj exists otherwise raise an error."""
+        if not self.proto_obj:
+            raise ValueError(
+                "There is no proto_obj!"
+                "\nUse create_new_intent or load_intent to continue."
+            )
+        elif not isinstance(self.proto_obj, Intent):
+            raise ValueError(
+                "proto_obj is not an Intent type."
+                "\nPlease create or load the correct type to continue."
+            )
+
+
+    def _show_basic_info(self) -> str:
+        """String representation for the basic information of proto_obj."""
+        self._check_intent_exist()
+
+        labels = [
+            str(key) if key == val else f"{key}: {val}"
+            for key, val in self.proto_obj.labels.items()
+        ]
+        return (f"display_name: {self.proto_obj.display_name}"
+            f"\ndescription: {self.proto_obj.description}"
+            f"\npriority: {self.proto_obj.priority}"
+            f"\nis_fallback: {self.proto_obj.is_fallback}"
+            f"\nlabels: {labels}")
+
+
+    def _show_parameters(self) -> str:
+        """String representation for the parameters of proto_obj."""
+        self._check_intent_exist()
+
+        return "\n".join([
+            (f"parameter_id: {str(param.id)}"
+            f"\nentity_type: {str(param.entity_type)}"
+            f"\n\tis_list: {bool(param.is_list)}"
+            f"\n\tredact: {bool(param.redact)}")
+            for param in self.proto_obj.parameters
+        ])
+
+
+    def _show_training_phrases(self, repeat_count: int = None) -> str:
+        """String representation for the training phrases of proto_obj."""
+        self._check_intent_exist()
+
+        phrases = []
+        for tp in self.proto_obj.training_phrases:
+            if repeat_count and tp.repeat_count != repeat_count:
+                continue
+            phrase = "".join([part.text for part in tp.parts])
+            annotations = {
+                part.text: part.parameter_id
+                for part in tp.parts
+                if part.parameter_id
+            }
+            phrases.append(
+                f"phrase: {phrase}"
+                f"\nannotations: {str(annotations)}"
+                f"\n\trepeat_count: {tp.repeat_count}"
+            )
+
+        return "\n".join(phrases)
+
+
+    def show_intent(
+        self, mode: str = "whole", repeat_count: int = None
+    ) -> None:
+        """Show the proto_obj information.
+
+        Args:
+          mode (str):
+            Specifies what part of the intent to show.
+              Options:
+              ['basic', 'parameters', 'phrases' or 'training phrases', 'whole']
```

Comment:
That's a great idea. We can differentiate between `phrases` and `training phrases`. maybe one with annotations and one without.
Or we can just leave them as they are since we discussed moving `intent_proto_to_dataframe` to this class, user can achieve the same results (show training phrases in "basic" and "advanced" mode).
As you mentioned, it'll be more clear to have just one of `phrases` or `training phrases` in Options so I'll remove `training phrases` if we decide to keep the function as it is.

Reply:
@MRyderOC I think what you did with the docstring is good

```
Options:
              ['basic', 'parameters', 'phrases' or 'training phrases', 'whole']
```

Basically saying `phrases` or `training phrases` indicates that either of these should return the same thing. 
I think you can leave it like this (with the `or` in between) and it will be good.